sdk/scheduler/src/main/java/com/mesosphere/sdk/offer/LoggingUtils.java

```diff
@@ -0,0 +1,51 @@
+package com.mesosphere.sdk.offer;
+
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Utility methods around construction of loggers.
+ */
+public class LoggingUtils {
+
+    private static boolean loggingNamesEnabled = true;
+
+    private LoggingUtils() {
+        // do not instantiate
+    }
+
+    /**
+     * Creates a logger which is tagged with the provided class.
+     *
+     * @param clazz the class using this logger
+     */
+    public static Logger getLogger(Class<?> clazz) {
+        // Note: This currently results in a name like "LoggingUtils" in logs.
+        // Consider including an abbreviated package prefix, e.g. "c.m.s.o.LoggingUtils"?
```

Comment:
I do think it is worth having an abbreviated prefix. We might have collisions otherwise.

Something like this (I can't pass up an opportunity to do lambdas...)

```
    public static Logger getLogger(Class<?> clazz) {
        // Note: This currently results in a name like "LoggingUtils" in logs.
        // Consider including an abbreviated package prefix, e.g. "c.m.s.o.LoggingUtils"?
        return getLogger(getAbbreviatedClassName(clazz));
    }

    /**
     * Creates a logger which is tagged with the provided class and the provided custom label.
     *
     * @param clazz the class using this logger
     * @param name  an additional context label detailing e.g. the name of the service being managed
     */
    public static Logger getLogger(Class<?> clazz, String name) {
        if (StringUtils.isEmpty(name) || !loggingNamesEnabled) {
            return getLogger(clazz);
        } else {
            return getLogger(String.format("(%s) %s", name, getAbbreviatedClassName(clazz)));
        }
    }

    private static Logger getLogger(String name) {
        return LoggerFactory.getLogger(name);
    }

    private static String getAbbreviatedClassName(Class<?> clazz) {
        String fullName = clazz.getName();
        List<String> nameParts = Arrays.asList(fullName.split("."));
        String prefix = nameParts.subList(0, nameParts.size())
                .stream()
                    .map(namePart -> namePart.substring(0, 1).toLowerCase())
                    .collect(Collectors.joining(","));

        return String.format("%s.%s", prefix, nameParts.get(nameParts.size() - 1));
    }

```

Reply:
It's definitely possible there exists something for this already...

Reply:
https://stackoverflow.com/questions/7891910/log4j-abbreviate-shorten-package-names

Hm, we can probably just configure our logger to do this.

Reply:
```
%c{1.}	org.apache.commons.Foo	results in o.a.c.Foo
```

Reply:
Are we using apache logging on the backend?

Reply:
Also, I'd be somewhat surprised that would work, as `LoggerFactory` itself hardcodes the label to `class.getName()`: https://www.slf4j.org/api/src-html/org/slf4j/LoggerFactory.html#line.379

Reply:
Oh hey, it works!

Reply:
So, with package vs without. Personally I'm still a little partial to without:

```
INFO  2018-03-14 17:26:06,728 [Test worker] c.m.s.s.p.DefaultPlanCoordinator:getCandidates(73): Updated dirtied assets: []
INFO  2018-03-14 17:26:06,728 [Test worker] c.m.s.s.p.DefaultPlanCoordinator:getCandidates(79): Got total candidates: [unreserve-e3e6d735-4dde-4b23-80ff-e73b1cd0de2f]
INFO  2018-03-14 17:26:06,728 [Test worker] c.m.s.s.AbstractScheduler:processQueuedOffers(429): InProgress Steps: []
INFO  2018-03-14 17:26:06,728 [Test worker] c.m.s.s.ReviveManager:revive(72): Candidates, old: [kill-world-1-server [N/A]], current: [unreserve-e3e6d735-4dde-4b23-80ff-e73b1cd0de2f [N/A]], new:[unreserve-e3e6d735-4dde-4b23-80ff-e73b1cd0de2f [N/A]]
WARN  2018-03-14 17:26:06,728 [Test worker] c.m.s.s.ReviveManager:revive(90): Revive attempt has been throttled.
INFO  2018-03-14 17:26:06,728 [Test worker] c.m.s.s.AbstractScheduler:processQueuedOffers(437): Processing 1 offer against 1 step:
INFO  2018-03-14 17:26:06,729 [Test worker] c.m.s.s.AbstractScheduler:processQueuedOffers(441):   1: AN OFFER HERE
INFO  2018-03-14 17:26:06,729 [Test worker] c.m.s.s.p.DefaultPlanScheduler:resourceOffers(71): Processing resource offers for step: unreserve-e3e6d735-4dde-4b23-80ff-e73b1cd0de2f
INFO  2018-03-14 17:26:06,729 [Test worker] c.m.s.s.u.ResourceCleanupStep:start(31): Setting state to Prepared for resource e3e6d735-4dde-4b23-80ff-e73b1cd0de2f
```

```
INFO  2018-03-14 17:26:06,728 [Test worker] DefaultPlanCoordinator:getCandidates(73): Updated dirtied assets: []
INFO  2018-03-14 17:26:06,728 [Test worker] DefaultPlanCoordinator:getCandidates(79): Got total candidates: [unreserve-e3e6d735-4dde-4b23-80ff-e73b1cd0de2f]
INFO  2018-03-14 17:26:06,728 [Test worker] AbstractScheduler:processQueuedOffers(429): InProgress Steps: []
INFO  2018-03-14 17:26:06,728 [Test worker] ReviveManager:revive(72): Candidates, old: [kill-world-1-server [N/A]], current: [unreserve-e3e6d735-4dde-4b23-80ff-e73b1cd0de2f [N/A]], new:[unreserve-e3e6d735-4dde-4b23-80ff-e73b1cd0de2f [N/A]]
WARN  2018-03-14 17:26:06,728 [Test worker] ReviveManager:revive(90): Revive attempt has been throttled.
INFO  2018-03-14 17:26:06,728 [Test worker] AbstractScheduler:processQueuedOffers(437): Processing 1 offer against 1 step:
INFO  2018-03-14 17:26:06,729 [Test worker] AbstractScheduler:processQueuedOffers(441):   1: AN OFFER HERE
INFO  2018-03-14 17:26:06,729 [Test worker] DefaultPlanScheduler:resourceOffers(71): Processing resource offers for step: unreserve-e3e6d735-4dde-4b23-80ff-e73b1cd0de2f
INFO  2018-03-14 17:26:06,729 [Test worker] ResourceCleanupStep:start(31): Setting state to Prepared for resource e3e6d735-4dde-4b23-80ff-e73b1cd0de2f
```

Here are all duplicate .java files in dcos-commons:

```
dcos-commons$ find . -iname *.java -exec basename {} \; | sort | uniq -c | grep -v "      1 "
      2 LaunchedTask.java
      8 Main.java
      6 ServiceTest.java
```
For `LaunchedTask`, there's one in `executor` (custom executor) and one in `testing` (service tests).

Reply:
Yeah, looking at the two side by side the abbreviated is kinda terrible looking.

I'm okay with just using `simpleName()` for now.

Reply:
Going with getSimpleName(). Now that we have the utility used everywhere, we can always revisit later